task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -1,20 +1,65 @@
+import threading
+import time
+from enum import auto
+from enum import unique
+
+from pyrsistent import field
+from pyrsistent import pmap
+from pyrsistent import PRecord
+from pyrsistent.typing import PMap
+
 from task_processing.interfaces import TaskExecutor
 from task_processing.plugins.kubernetes.task_config import KubernetesTaskConfig
+from task_processing.utils import AutoEnum
+
+
+@unique
+class KubernetesTaskState(AutoEnum):
+    TASK_PENDING = auto()
+
+
+class KubernetesTaskMetadata(PRecord):
+    # what box this task/Pod was scheduled onto
+    node_name: str = field(type=str, initial='')
+
+    # the config used to launch this task/Pod
+    task_config: KubernetesTaskConfig = field(type=KubernetesTaskConfig, mandatory=True)
+
+    # TODO(TASKPROC-241): add current task state and task state history as we did for mesos
+    task_state: KubernetesTaskState = field(type=KubernetesTaskState, mandatory=True)
+    # Map of state to when that state was entered (stored as a timestamp)
+    task_state_history: PMap[str, int] = field(factory=pmap, mandatory=True)
```

Comment:
it's like this in mesos, but i think using map was a mistake, list of (timestamp, state) would preserve history correctly in case when task bounces between same states. might be good chance to fix this.

Reply:
I'd actually had that thought but assumed that we had done it that way on purpose in the mesos plugin!

fixed :)